### ISSUE TITLE
.css file not bundled while using builtin:swc-loader

### ISSUE THREAD

**Issue by saidarshan27** [web-infra-dev/rspack#8167]
I am using a package called `plyr-react`  for video player support in an application. `plyr-react` exports a `plyr.css` file, which is not being included in my final bundle even after being referenced in the entry file. 

Here is my rspack config

```
module.exports = {
  mode: "development",
  entry: './src/javascripts/index/index.js',
  output: {
    filename: "main.[contenthash].js",
    path: path.resolve(__dirname, "dist")
  },
  plugins: [
    new rspack.CssExtractRspackPlugin({
      filename: '[name].[contenthash].css',
      chunkFilename: '[id].[name].[contenthash].css'
    }),
    new HTMLWebpackPlugin({
      filename: 'index.html',
      template: './index.html',
    }),
    new CleanWebpackPlugin()
  ],
  module: {
    rules: [
      {
        test: /\.css$/,
        use: [rspack.CssExtractRspackPlugin.loader, "css-loader"],
      },
      {
        test: /\.scss$/,
        use: [rspack.CssExtractRspackPlugin.loader, "css-loader", "sass-loader"]
      },
      {
        test: /\.js(x)?$/,
        exclude:
          /node_modules/,
        use: [
          {
            loader: 'builtin:swc-loader',
            options: {
              jsc: {
                parser: {
                  syntax: "ecmascript",
                  jsx: true,
                },
                transform: {
                  react: {
                    pragma: 'React.createElement',
                    pragmaFrag: 'React.Fragment',
                    throwIfNamespace: true,
                    development: false,
                    useBuiltins: false
                  }
                },
              },
            }
          },
        ],
      }
    ]
  },
  optimization: {
    splitChunks: {
      chunks: 'all',
      name: false,
      maxInitialRequests: 25,
      maxAsyncRequests: 25,
    },
  },
}; 
```

And here you can see me referencing `plyr.css` along with `css-component.css`. 

![Image](https://github.com/user-attachments/assets/ec96bdd6-4424-4095-8c74-3adc57835f67)

Here are the bundled files

![Image](https://github.com/user-attachments/assets/49b5a22a-6b1d-4e01-8558-32e5af4a4d4e)

And here you can see the bundled `main.css`, which does not include CSS code from `plyr.css` 

![Image](https://github.com/user-attachments/assets/366933a2-8739-4e90-86c4-b9f5d9c4b798)

 This issue does not occur when I use `babel-loader`.

**Comment by saidarshan27** [comment]
Node - v18.12.0;
yarn - 1.22.19;
@rspack/core - 1.0.11;
@rspack/cli - 1.0.8;

Here is a reproduction demo:
https://github.com/saidarshan27/swc-plyr-css

**Comment by rotate-life** [comment]
try it

**Comment by rotate-life** [comment]
try it 
`minimizer: [new rspack.LightningCssMinimizerRspackPlugin({
      removeUnusedLocalIdents: false,
   })]`

**Comment by saidarshan27** [comment]
no, still the same. 

actually the reproduction demo is a dumbed down version of the code. 
 
In the actual codebase, the class definitions from `plyr.css` are being used. 

Let me also add that, if I import the `plyr.css` inside the `css-component.css`. The styles are being loaded properly, only when I import it directly in `index.js`, styles aren't being included in the bundled `main.css`.

**Comment by inottn** [comment]
This might be related to the value of `sideEffects` in the package.json.

**Comment by saidarshan27** [comment]
Yes @inottn , it is related to `sideEffects`. I removed the `sideEffects` from `package.json` of `plyr-react`. 
Now the styles are being bundled properly. 
But what changes should I make in my rspack configuration such that, the styles are bundled even though the `sideEffects` is `false`

**Comment by inottn** [comment]
In fact, the `sideEffects` here should not be set to `false` because your CSS imports have side effects. You can assign the correct value to `sideEffects` based on the actual situation.

**Comment by saidarshan27** [comment]
The problem is I cannot change the `package.json` of the package. It is a third party package that we are using.

This problem does not occur when I use `babel-loader` instead of `swc-loader`. So I was thinking if there was anything I was missing in the rspack configuration that would fix this? @inottn

**Comment by saidarshan27** [comment]
I tried `optimization.sideEffects: true`  in my configuration but it does not fix the problem. 
I wanted why this problem is occurring when I use `builtin:swc-loader` and not occurring when I use `babel-loader`.

**Comment by inottn** [comment]
Try setting optimization.sideEffects: false?

**Comment by saidarshan27** [comment]
yes setting it to `optimization.sideEffects: false` does the job. 
But does this any how affect the tree shaking process for other modules?

**Comment by LingyuCoder** [comment]
According to your repro, the css is only imported and no export item is used. So it should be removed by tree shaking because the `sideEffects` is `false` in `package.json` of `plyr-react`. It is recommend to set `sideEffects` to these files by `module.rules`, just modify your css rule like this:

```js
// before
{
  test: /\.css$/,
  use: [rspack.CssExtractRspackPlugin.loader, "css-loader"],
},

// after
{
  test: /\.css$/,
  exclude: /plyr-react/,
  use: [rspack.CssExtractRspackPlugin.loader, "css-loader"],
},
{
  test: /\.css$/,
  include: /plyr-react/,
  use: [rspack.CssExtractRspackPlugin.loader, "css-loader"],
  sideEffects: true,
},
```
